Migration ProConnect

ikarius opened the pull request:
ToDo ...

# Migration ProConnect : frontend

Cette PR n'est pas prête pour la production car elle contient les accès aux 2 systèmes d'identification (IC et PC).

Assez peu de choses sur la partie frontend, la plupart des actions du flow OIDC étant effectuées coté backend  (voir https://github.com/gip-inclusion/dora-back/pull/325).

ikarius committed:
Ajout du bouton ProConnect officiel

ikarius committed:
oidc: redirection après callback d'identification OIDC

ikarius committed:
oidc: effacement des données de connexion avant logout OIDC

En 2 phases :
- effacement du token coté frontend
- finalisation de la déconnexion coté backend

ikarius committed:
Ajout du bouton de déconnexion ProConnect

ggounot commented on src/app.postcss:
```diff
@@ -172,4 +172,30 @@
+
+  .proconnect-button {
+    background-color: transparent !important;
+    background-image: url("data:image/svg+xml;base64,PHN2ZyB4bWxucz0naHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmcnIHdpZHRoPScyMTEnIGhlaWdodD0nNTgnIGZpbGw9J25vbmUnPjxwYXRoIGZpbGw9JyMwMDAwOTEnIGQ9J00wIDBoMjExdjU4SDB6Jy8+PHBhdGggZmlsbD0nI2ZmZicgZD0nbTY5Ljk4NiAyNi4zNjggMS4xNTYtMS4wNzFjLjgzMyAxLjA1NCAxLjgxOSAxLjU5OCAyLjk0MSAxLjU5OCAxLjI5MiAwIDIuMDQtLjgxNiAyLjA0LTEuOTA0IDAtMi41NS01LjYyNy0yLjI0NC01LjYyNy02LjAzNS ...
```
Personnellement pas fan des images encodées dans le CSS, en particulier lorsqu'elles résultent en de longues chaînes base64. Peut-on en faire des fichiers à part entière ?

ikarius committed:
Ajout d'une variable d'environnement pour identifier le backend OIDC

On peut encore choisir entre Inclusion Connect et ProConnect via la
var-env OIDC_AUTH_BACKEND (par défaut `proconnect`)
